src/openms/source/ANALYSIS/ID/IDMapper.cpp

```diff
@@ -322,12 +322,41 @@ namespace OpenMS
           //check if we compare distance from centroid or subelements
           if (!measure_from_subelements)
           {
-            if (isMatch_(rt_pep - map[cm_index].getRT(), mz_pep, map[cm_index].getMZ()) && (ignore_charge_ || ListUtils::contains(current_charges, map[cm_index].getCharge())))
+            if (map[cm_index].metaValueExists("id_scan_id")) // MS3 TMT
```

Comment:
Code seems to be duplicated 
- maybe first set MS3, MS2 or label-free in separate function. 
- set reference id once (either id_scan_id or scan_id)
- and use same code to push back ids?
especially as oder of if evaluation is critical here and not documented